cmd/spiffe-aws-assume-role/cli/cli.go

```diff
@@ -34,13 +37,15 @@ func (c *CredentialsCmd) Run(context *CliContext) error {
 
 	src := context.JWTSourceProvider(spiffeID, c.WorkloadSocket, c.Audience)
 
+	session := createSession(c.STSEndpoint, c.STSRegion)
+	stsClient := context.STSProvider(session)
+
 	provider, err := credentials.NewProvider(
 		c.Audience,
 		c.RoleARN,
 		src,
 		c.SessionDuration,
-		context.STSProvider,
-		c.STSEndpoint)
+		stsClient)
 	if err != nil {
 		return err
 	}
```

Comment:
We're changing the Provider constructor to take in a fully-configured STS client (which is a pretty core principle of IoC so shame on me). This means that the various construction logic (Session, etc.) gets pulled up a lever. On the plus side, the constructor takes in one less parameter.